arviz/plots/backends/matplotlib/pairplot.py

```diff
@@ -39,13 +44,48 @@ def plot_pair(
         **backend_kwarg_defaults(),
         **backend_kwargs,
     }
+    backend_kwargs.pop("constrained_layout")
+
     if numvars == 2:
-        (figsize, ax_labelsize, _, xt_labelsize, _, _) = _scale_fig_size(
+        (figsize, ax_labelsize, _, xt_labelsize, linewidth, _) = _scale_fig_size(
             figsize, textsize, numvars - 1, numvars - 1
         )
 
-        if ax is None:
-            fig, ax = plt.subplots(figsize=figsize, **backend_kwargs)
+        if marginal_kwargs is None:
+            marginal_kwargs = {}
+
+        marginal_kwargs.setdefault("plot_kwargs", {})
+        marginal_kwargs["plot_kwargs"]["linewidth"] = linewidth
+
+        # Flatten data
+        x = infdata_group[0].flatten()
+        y = infdata_group[1].flatten()
+
+        if diagonal:
+            # Instantiate figure and grid
+            fig, _ = plt.subplots(0, 0, figsize=figsize, **backend_kwargs)
+            grid = plt.GridSpec(4, 4, hspace=0.1, wspace=0.1, figure=fig)
+
+            if ax is None:
+                # Set up main plot
+                ax = fig.add_subplot(grid[1:, :-1])
+            # Set up top KDE
+            ax_hist_x = fig.add_subplot(grid[0, :-1], sharex=ax)
+            # Set up right KDE
+            ax_hist_y = fig.add_subplot(grid[1:, -1], sharey=ax)
+
+            for val, ax_, rotate in ((x, ax_hist_x, False), (y, ax_hist_y, True)):
+                plot_dist(val, textsize=xt_labelsize, rotated=rotate, ax=ax_, **marginal_kwargs)
+
+            ax_hist_x.set_xlim(ax.get_xlim())
+            ax_hist_y.set_ylim(ax.get_ylim())
+
+            # Personalize axes
+            ax_hist_x.tick_params(labelleft=False, labelbottom=False)
+            ax_hist_y.tick_params(labelleft=False, labelbottom=False)
+        else:
+            if ax is None:
+                fig, ax = plt.subplots(figsize=figsize, **backend_kwargs)
 
         if kind == "scatter":
```

Comment:
how about reusing the code for scatter and kde? Something like:

```
if "scatter" in kind:
...
if "kde" in kind: # not elif
...
if "hexbin" in kind:
...
```

For now the valid kinds do not allow for more alternatives, but this is easily extended to a hexbin plot covering the scatter plot so only the point further from the mean are seen (in a corner.py style fashion)

Reply:
I am thinking that if multiple kinds can take place in the same plot, `plot_kwargs` will eventually present kwarg incompatibilities. Maybe we should have `scatter_kwargs`, `kde_kwargs` and `hexbin_kwargs`. Or just `plot_kwargs` and find a way to distribute key words.

Reply:
Having separated `kwargs` is much more flexible for the user and also easier to maintain than having to properly parse a `plot_kwargs` and distribute its content .